SplitTests/Integration/streaming/StreamingControlTest.swift

```diff
@@ -0,0 +1,212 @@
+        splitConfig.featuresRefreshRate = kRefreshRate
+        splitConfig.segmentsRefreshRate = kRefreshRate
+        splitConfig.impressionRefreshRate = 30
+        splitConfig.sdkReadyTimeOut = 60000
+        splitConfig.eventsPerPush = 10
+        splitConfig.eventsQueueSize = 100
+        splitConfig.eventsPushRate = 5
+
+        let splitName = "workm"
+
+        let key: Key = Key(matchingKey: userKey)
+        let builder = DefaultSplitFactoryBuilder()
+        _ = builder.setHttpClient(httpClient)
+        _ = builder.setReachabilityChecker(ReachabilityMock())
+        let factory = builder.setApiKey(apiKey).setKey(key)
+            .setConfig(splitConfig).build()!
+
+        var timestamp = 1000
+
+        let client = factory.client
+
+        let sdkReadyExpectation = XCTestExpectation(description: "SDK READY Expectation")
+        createExpectations()
+
+        client.on(event: SplitEvent.sdkReady) {
+            sdkReadyExpectation.fulfill()
+        }
+
+        client.on(event: SplitEvent.sdkReadyTimedOut) {
+            sdkReadyExpectation.fulfill()
+        }
+
+        wait(for: [sdkReadyExpectation, sseExp], timeout: 20)
+
+        let treatmentReady = client.getTreatment(splitName)
+
+        justWait() // wait to allow sync all
+        
+        // Should disable streaming
+        timestamp+=1000
+        streamingBinding?.push(message: StreamingIntegrationHelper.controlMessage(timestamp: timestamp,
+                                                                                  controlType: "STREAMING_PAUSED"))
+        waitForHits()
+        wait(for: [mySegExps[mySegExpIndex],  splitsChangesExps[splitsExpIndex]], timeout: 7)
+
+        timestamp+=1000
+        streamingBinding?.push(message: StreamingIntegrationHelper.mySegmentWithPayloadMessage(timestamp: timestamp,
+                                                                                               segment: "new_segment"))
+        justWait() // allow to my segments be updated
+        let treatmentPaused = client.getTreatment(splitName)
+
+
+        timestamp+=1000
+        streamingBinding?.push(message: StreamingIntegrationHelper.controlMessage(timestamp: timestamp,
+                                                                                  controlType: "STREAMING_ENABLED"))
+
+        timestamp+=1000
+        streamingBinding?.push(message: StreamingIntegrationHelper.mySegmentWithPayloadMessage(timestamp: timestamp,
+                                                                                               segment: "new_segment"))
+        justWait() // allow to my segments be updated
+        let treatmentEnabled = client.getTreatment(splitName)
+
+
+        timestamp+=1000
+        streamingBinding?.push(message: StreamingIntegrationHelper.controlMessage(timestamp: timestamp,
+                                                                                  controlType: "STREAMING_DISABLEd"))
```

Comment:
Javi check that "STREAMING_DISABLEd" is misspelled. So I think the test should be actually failing.

Reply:
Don't remember if it is converting the string to lowercase letters. In that case, it is fine.

Reply:
Doc, thanks for the comment. When the control notification is parsed the value is lowercased to avoid any issue. The test shouldn't fail for this reason.